code/modules/projectiles/guns/syringe_gun.dm

```diff
@@ -95,3 +94,249 @@
 	force = 2 //Also very weak because it's smaller
 	suppressed = 1 //Softer fire sound
 	can_unsuppress = 0 //Permanently silenced
+
+// Not quite a syringe gun, but also not completely unlike one either.
+// these names are subject to change, I just want to leave stuff in place for now.
+/obj/item/gun/syringe/rapidsyringe
+	name = "rapid syringe gun"
+	desc = "A syndicate rapid syringe gun stolen from NanoTrasen's R&D archives. Has an internal mechanism capable of storing and filling syringes from an internal reservoir.\nIt has a large flap on the side that you can dump a box or bag of syringes into, and a port for filling it with liquid."
+	icon_state = "rapidsyringegun"
+	max_syringes = 14  // full two boxes worth
+	/// Internal reagent container.
+	var/obj/item/reagent_containers/glass/beaker/bluespace/internal_beaker
+	/// Index of possible transfer amounts to select
+	var/reagents_per_syringe_selection = 1
+	/// Possible options for alt-clicking the
+	var/list/possible_transfer_amounts = list(5, 10, 15)
```

Comment:
I feel like these vars should have their names standardised. so either have `possible_transfer_amounts` and `transfer_amount_selection` or `possible_reagents_transfered_per_syringe` and `reagents_per_syringe_selection`

Personally prefer the first one, fewer words, conveys the same information.